### PR TITLE
Add Slider

### PR DESCRIPTION
**Pull request by JaneEllison**
Add slider component. Add slider for Favorite cities, daily forecast components. Add changing current city by Favorite cities card click. Fix styles.

### CODE REVIEW

File: src/components/Slider/Slider.module.scss [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    justify-content: center;
+    position: absolute;
+    right: -50px;
+    height: 50px;
+    top: 50%;
+    transform: translate(0, -50%);
+    width: 50px;
+    border: 0;
+    z-index: 1;
+    font-size: 34px;
```
Comment (glebdronchenko) [inline src/components/Slider/Slider.module.scss]: Hardcoded value.

File: src/components/Slider/Slider.module.scss [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    background-color: $primary-bg-color;
+    color: $arrow-btn-color;
+    cursor: pointer;
+    outline: none;
+    transition: font-size $default-transition-delay ease;
+
+    &:hover {
+      color: $hover-arrow-btn-color;
+      font-size: 40px;
```
Comment (glebdronchenko) [inline src/components/Slider/Slider.module.scss]: Hardcoded value.

File: src/components/Slider/Slider.module.scss (hunk 1 above)
Comment (JaneEllison) [reply src/components/Slider/Slider.module.scss]: Thank you. Fixed

File: src/components/Slider/Slider.module.scss (hunk 2 above)
Comment (JaneEllison) [reply src/components/Slider/Slider.module.scss]: Thank you. Fixed

File: src/components/SearchInput/SearchInput.module.scss [hunk 3]
```diff
@@ -90,6 +91,7 @@
 
 .search-wrapper:hover .search-results {
   opacity: 1;
+  z-index: 15;
```
Comment (anton-vlasik) [inline src/components/SearchInput/SearchInput.module.scss]: Why -5, 15 - how do you decide what values to assign?

File: src/redux/selectors/locationSelectors.js [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
 export const getCityInfo = state => state.location.currentCity;
+const getFavoriteCities = state => state.location.recentCities;
 
 export const getCityTitle = createSelector([getCityInfo], ({ country, name, id }) => ({
   country,
   name,
   id
 }));
+
+export const getCityCards = createSelector([getFavoriteCities], favoriteCities => {
```
Comment (anton-vlasik) [inline src/redux/selectors/locationSelectors.js]: this seems to be a selector and I suggested you to name them differently from simple getters

File: src/redux/selectors/locationSelectors.js [hunk 5]
```diff
@@ -1,9 +1,18 @@
...
+  return favoriteCities.map(city => ({
+    id: city.id,
+    slide: <CityCard info={city} />
```
Comment (anton-vlasik) [inline src/redux/selectors/locationSelectors.js]: This time of mapping usually just happen in render function, so it is either useless or - did you just mean to 
```
export const getCityCards = createSelector(getFavoriteCities, favoriteCities => {
     return favoriteCities.map(city => ({ id: city.id, slideInfo: city }));
});
```
which then again makes it pretty useless as city already contains id

File: src/redux/selectors/weatherSelectors.js [hunk 6]
```diff
@@ -18,3 +20,10 @@ export const getShortCityForecast = createSelector(
     windSpeed
   })
 );
+
+export const getDailyForecastCards = createSelector([getDailyForecast], dailyForecast => {
+  return dailyForecast.map(forecast => ({
+    id: forecast.date,
+    slide: <WeatherCard weatherInfo={forecast} />
```
Comment (anton-vlasik) [inline src/redux/selectors/weatherSelectors.js]: same here.

File: src/components/SearchInput/SearchInput.module.scss (hunk 3 above)
Comment (JaneEllison) [reply src/components/SearchInput/SearchInput.module.scss]: I rewiew all z-indexes and remove excess.  
What about value step  - I am thought about situation when we need to add element with z-index between this two value. And in this case we need to rewrite all indexes and add new index. And if we have value step equal to 5, we can add any value between. but these are situations that rarely happend)

File: src/redux/selectors/locationSelectors.js (hunk 4 above)
Comment (JaneEllison) [reply src/redux/selectors/locationSelectors.js]: renamed to selectCityCards

File: src/redux/selectors/locationSelectors.js (hunk 5 above)
Comment (JaneEllison) [reply src/redux/selectors/locationSelectors.js]: I create an array with CityCard components and pass this array to Slider component where i render those components, thats why i need id field. But I refactored this piece for better understanding - rename property name, remove id property  - now I passed only array of components. And also I added library for random ids in Slider component.

File: src/redux/selectors/weatherSelectors.js (hunk 6 above)
Comment (JaneEllison) [reply src/redux/selectors/weatherSelectors.js]: I refactored it.

**Review by anton-vlasik — APPROVED** [review]

**Pull request merged**